Userland/Libraries/LibWeb/Loader/GeneratedPagesLoader.cpp

```diff
@@ -33,6 +33,7 @@ ErrorOr<String> load_error_page(URL::URL const& url, StringView error_message)
     auto template_file = TRY(Core::Resource::load_from_uri("resource://ladybird/templates/error.html"sv));
     StringBuilder builder;
     SourceGenerator generator { builder };
+    generator.set("media", "@media"sv);
```

Comment:
This is a bit of a hack, but I couldn't see a way to avoid SourceGenerator from thinking it's a substitution value.

Reply:
How about modifying SourceGenerator, so that you can escape the opening character with a backslash? I think something like this should work:

```diff
diff --git a/AK/SourceGenerator.h b/AK/SourceGenerator.h
index 7a2c0e444a..28265018a6 100644
--- a/AK/SourceGenerator.h
+++ b/AK/SourceGenerator.h
@@ -21,17 +21,19 @@ class SourceGenerator {
 public:
     using MappingType = HashMap<StringView, String>;

-    explicit SourceGenerator(StringBuilder& builder, char opening = '@', char closing = '@')
+    explicit SourceGenerator(StringBuilder& builder, char opening = '@', char closing = '@', char escape = '\\')
         : m_builder(builder)
         , m_opening(opening)
         , m_closing(closing)
+        , m_escape(escape)
     {
     }
-    explicit SourceGenerator(StringBuilder& builder, MappingType&& mapping, char opening = '@', char closing = '@')
+    explicit SourceGenerator(StringBuilder& builder, MappingType&& mapping, char opening = '@', char closing = '@', char escape = '\\')
         : m_builder(builder)
         , m_mapping(move(mapping))
         , m_opening(opening)
         , m_closing(closing)
+        , m_escape(escape)
     {
     }

@@ -70,7 +72,20 @@ public:
         GenericLexer lexer { pattern };

         while (!lexer.is_eof()) {
-            m_builder.append(lexer.consume_until(m_opening));
+            m_builder.append(lexer.consume_until([&](char ch) { return ch == m_opening || ch == m_escape; }));
+            if (lexer.consume_specific(m_escape)) {
+                if (!(lexer.next_is(m_opening) || lexer.next_is(m_escape))) {
+                    if (lexer.is_eof())
+                        warnln("Unexpected EOF while parsing escape sequence on SourceGenerator");
+                    else
+                        warnln("Invalid escape sequence found `{}{}` on SourceGenerator", m_escape, lexer.peek());
+
+                    VERIFY_NOT_REACHED();
+                }
+
+                m_builder.append(lexer.consume());
+                continue;
+            }

             if (lexer.consume_specific(m_opening)) {
                 auto const placeholder = lexer.consume_until(m_closing);
@@ -129,7 +144,9 @@ public:
 private:
     StringBuilder& m_builder;
     MappingType m_mapping;
-    char m_opening, m_closing;
+    char m_opening { '@' };
+    char m_closing { '@'};
+    char m_escape { '\\' };
 };
 ```

Reply:
I've applied this commit, seems a good change to have. 

In terms of resolving this hack though, I just changed the character used by the templates to be '%' rather than '@' (not sure why I didn't think to do this earlier :sweat_smile:).